ingest/src/main/java/com/microsoft/azure/kusto/ingest/AzureStorageHelper.java

```diff
@@ -46,51 +53,94 @@ void azureTableInsertEntity(String tableUri, TableServiceEntity entity) throws S
     CloudBlockBlob uploadLocalFileToBlob(String filePath, String blobName, String storageUri) throws URISyntaxException, StorageException, IOException {
         log.debug("uploadLocalFileToBlob: filePath: {}, blobName: {}, storageUri: {}", filePath, blobName, storageUri);
 
+        // Validation
+        validateIsNotEmpty(filePath, "filePath is empty");
+        validateIsNotEmpty(blobName, "blobName is empty");
+        validateIsNotEmpty(storageUri, "storageUri is empty");
+
+        File sourceFile = new File(filePath);
+        validateFileExists(sourceFile, "The file does not exist: " + filePath);
```

Comment:
If we don't validate it here, then it will continue to uploadFileToBlob() -> blob.uploadFromFile() -> uploadFromFile() -> And only here it will fail when it tries to create a FileInputStream and will throw an IOException.
Don't you think the validation must be done in the beginning?

Reply:
Will the exception say that the file doesn't exist?